enterprise_subsidy/apps/api/v1/views.py

```diff
@@ -1 +1,216 @@
+        return permission_required
+
+    @property
+    def requested_enterprise_customer_uuid(self):
+        """
+        Look in the query parameters for an enterprise customer UUID.
+        """
+        return utils.get_enterprise_uuid_from_request_query_params(self.request)
+
+    @property
+    def requested_subsidy_uuid(self):
+        """
+        Fetch the subsidy UUID from the URL location.
+
+        For detail endpoints, the PK can simply be found in ``self.kwargs``.
+        """
+        return self.kwargs.get('uuid')
+
+    @property
+    def base_queryset(self):
+        """
+        Required by the ``PermissionRequiredForListingMixin``.
+        For non-list actions, this is what's returned by ``get_queryset()``.
+        For list actions, some non-strict subset of this is what's returned by ``get_queryset()``.
+        """
+        kwargs = {}
+        if self.requested_enterprise_customer_uuid:
+            kwargs.update({'enterprise_customer_uuid': self.requested_enterprise_customer_uuid})
+        if self.requested_subsidy_uuid:
+            kwargs.update({'uuid': self.requested_subsidy_uuid})
+
+        return Subsidy.objects.filter(**kwargs).prefetch_related(
+            # Fields used for calculating the ledger balance.
+            'ledger__transactions__state',
+            'ledger__transactions__quantity',
+            'ledger__transactions__reversal',
+            'ledger__transactions__reversal__state',
+            'ledger__transactions__reversal__quantity',
+        ).order_by('uuid')
+
+    def create(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Create action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
+
+    def update(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Update action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
+
+    def partial_update(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Update action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
+
+    def destroy(self, *args, **kwargs):
+        """
+        We should never get to this point because the permission for this action is not granted, but just in case we'll
+        force a 403 response.
+        """
+        response = {'message': 'Destroy action is not offered.'}
+        return Response(response, status=status.HTTP_403_FORBIDDEN)
```

Comment:
I like the belt+suspenders approach, here.  Another thing worth considering as an addition to edx-rbac.

Reply:
On the other hand, DRF does provide: https://www.django-rest-framework.org/api-guide/viewsets/#readonlymodelviewset which I think is actually what we want, here.